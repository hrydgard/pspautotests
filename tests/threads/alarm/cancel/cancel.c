#include "../sub_shared.h"
#include <limits.h>

volatile int alarmHandlerHits = 0;

void testCancel(const char *title, SceUID alarm) {
	int result = sceKernelCancelAlarm(alarm);
	printf("%s: %08X\n", title, result);
}

static SceUInt alarmHandler(void *common) {
	alarmHandlerHits++;
	if (common != NULL) {
		return *(int *) common;
	} else {
		return 0;
	}
}

int main(int argc, char **argv) {
	SceUID alarm = sceKernelSetAlarm(1000, alarmHandler, NULL);

	testCancel("Normal", alarm);
	testCancel("Deleted", alarm);
	testCancel("Invalid", 0xDEADBEEF);
	testCancel("NULL", 0);

	alarm = sceKernelSetAlarm(10000, alarmHandler, NULL);
	BASIC_SCHED_TEST("Normal",
		result = sceKernelCancelAlarm(alarm);
	);
	BASIC_SCHED_TEST("Deleted",
		result = sceKernelCancelAlarm(alarm);
	);

	alarm = sceKernelSetAlarm(100, alarmHandler, NULL);
	sceKernelDelayThread(1000);
	testCancel("Already ran", alarm);

	return 0;
}