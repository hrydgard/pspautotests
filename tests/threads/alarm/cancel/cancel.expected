Normal: 00000000
Deleted: 8002019F
Invalid: 8002019F
NULL: 8002019F
Normal: A1B1C2E1D1F2 (thread=800201B5, main=00000000)
Deleted: A1B1C2E1D1F2 (thread=800201B5, main=8002019F)
Already ran: 8002019F
